### PR TITLE
Visualize in hierarchical order Hubs and Organizations

### PR DESCRIPTION
**Pull request by mcespedeso**

### CODE REVIEW

File: src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java [hunk 1]
```diff
@@ -22,95 +25,120 @@
...
-	    }
-        
+    }
+
+    @Override
+    public ApplicationDTO getApplicationDashboard(Long applicationId,
+            UserDetailsDTO details) {
+        ApplicationDTO dto = new ApplicationDTO();
+
+        if (details.getApplication() != null
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java]: Can we move this to a new method `getUserApplication()`? 
```java
// logic of the method
if (somethingIsTrue)
  return getApplicationById()
```

File: src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/OrganizationServiceImpl.java [hunk 2]
```diff
@@ -27,114 +27,129 @@
...
+                    });
+        }
+
+        return null;
+    }
+
+    @Override
+    public OrganizationDTO getOrganizationDashboard(Long organizationId,
+            UserDetailsDTO details) {
+        OrganizationDTO dto = new OrganizationDTO();
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/OrganizationServiceImpl.java]: Same here, move this logic of getting the organization to a new method.

File: src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/OrganizationServiceImpl.java [hunk 3]
```diff
@@ -27,114 +27,129 @@
...
-	@Override
-    public OrganizationDTO getOrganizationDashboard(Long organizationId, UserDetailsDTO details) {
-	    OrganizationDTO dto = new OrganizationDTO();
-        
-        if(details.getOrganization() != null && details.getOrganization().getId() != null) {
+
+    }
+
+    @Override
+    public Page<OrganizationDTO> listOrganizations(PageRequest pageRequest,
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/OrganizationServiceImpl.java]: This method should return a `PaginableList<ApplicationDTO>`

File: src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/OrganizationServiceImpl.java [hunk 4]
```diff
@@ -27,114 +27,129 @@
...
+            Long applicationId, Long organizationId) {
+
+        Page<OrganizationEntity> pageResponse = organizationRepository.findAll(
+                where(byFilter(applicationId, organizationId)), pageRequest);
+
+        if (pageResponse != null) {
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/OrganizationServiceImpl.java]: Prefer the use of early returns

```java
if (pageResponse == null) {
    return new PaginableList(Collections.emptyList());
}
// here is safe to use pageResponse
```

File: src/main/java/py/org/fundacionparaguaya/pspserver/web/rest/ApplicationController.java [hunk 5]
```diff
@@ -20,61 +22,90 @@
...
 
+    @GetMapping("/hubs")
+    public ResponseEntity<PaginableList<ApplicationDTO>> getAllApplicationsHubs(
+            @RequestParam(value = "page", required = false, defaultValue = "1") int page,
+            @RequestParam(value = "per_page", required = false, defaultValue = "12") int perPage,
+            @RequestParam(value = "sort_by", required = false, defaultValue = "name") String sortBy,
+            @RequestParam(value = "order", required = false, defaultValue = "asc") String orderBy) {
+        PageRequest pageRequest = new PspPageRequest(page, perPage, orderBy,
+                sortBy);
+        Page<ApplicationDTO> pageProperties = applicationService
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/web/rest/ApplicationController.java]: This logic should be moved to the service.

**Comment by mcespedeso** [comment]
@rodrigojv, All modifications has been made! #113

File: src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java [hunk 6]
```diff
@@ -28,117 +31,133 @@
...
-        dto.setDashboard(
+    dto.setDashboard(
                 DashboardDTO.of(familyService.countFamiliesByFilter(filter)));
 
-        return dto;
+    return dto;
+  }
+
+  @Override
+  public ApplicationDTO getUserApplication(UserDetailsDTO details,
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java]: If we don't use this method outside the service, should be private.

File: src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java [hunk 7]
```diff
@@ -28,117 +31,133 @@
...
+                sortBy);
+
+    Page<ApplicationEntity> pageResponse = applicationRepository
                 .findAllByIsHub(true, pageRequest);
 
-        if (pageResponse != null) {
-            return pageResponse
+    if (pageResponse == null) {
+      return new PaginableList<>(Collections.emptyList());
+    } else {
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java]: we don't need this else.
```java
if (page==null)  {
    return  something;
}
// more logic
return new PapginableList(..., ...);
```

**Comment by mcespedeso** [comment]
@rodrigojv modifications made!

File: src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java [hunk 8]
```diff
@@ -31,133 +30,130 @@
...
+    }
+
+    @Override
+    public PaginableList<ApplicationDTO> listApplicationsHubs(int page,
+                    int perPage, String orderBy, String sortBy) {
+
+        PageRequest pageRequest = new PspPageRequest(page, perPage, orderBy,
+                        sortBy);
+
+        Page<ApplicationEntity> pageResponse = applicationRepository
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java]: Where did the null check go? Is it safe to do a pageResponse.map without null checking?

File: src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java [hunk 9]
```diff
@@ -139,21 +140,29 @@ private ApplicationDTO getUserApplication(UserDetailsDTO details,
...
-                                            ApplicationEntity source) {
-                                return applicationMapper.entityToDto(source);
-                            }
-                        });
-
-        return new PaginableList<ApplicationDTO>(applicationPage,
-                        applicationPage.getContent());
+        if (pageResponse == null) {
+            return new PaginableList<>(Collections.emptyList());
+          } else {
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java]: Preguntemos por null pero no pongamos la palabra `else`.
```java
if (pageResponse == null) {
  return ....
}
// aca va el resto del codigo donde es seguro usar pageResponse pero no hace falta escribir else.
```
Se entiende?

Comment (rodrigojv) [reply src/main/java/py/org/fundacionparaguaya/pspserver/network/services/impl/ApplicationServiceImpl.java]:
```java
if (pageResponse == null) {
   return  new PaginableList<ApplicationDTO>(Collections.emptyList();
}
 Page<ApplicationDTO> applicationPage = pageResponse.map(
      			  new Converter<ApplicationEntity, ApplicationDTO>() {
         				  @Override
         				  public ApplicationDTO convert(
         						  ApplicationEntity source) {
         					  return applicationMapper.entityToDto(source);
         				  }
         			  });
return new PaginableList<ApplicationDTO>(applicationPage,
    			  applicationPage.getContent());
```

**Pull request merged**